### PR TITLE
feat: add R.modify type definition for @types/ramda

### PR DESCRIPTION
**Pull request by adajoy**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://ramdajs.com/docs/#modify>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ramda ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ramda@latest/))
was missing the following properties:
1. modifyPath
1. sequence
1. unwind
1. whereAny
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0057a6e247f3297e1172a420924bc948738b5c5b
</p>

File: types/ramda/index.d.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@
 //                 Nikita Balikhin <https://github.com/NEWESTERS>
 //                 Wang Zengdi <https://github.com/adispring>
 //                 Marcus Blättermann <https://github.com/essenmitsosse>
+//                 adajoy <https://github.com/adajoy>
```
Comment (jakebailey) [inline types/ramda/index.d.ts:34]: Are you adding yourself intentionally? This adds you as an owner of the package; it's not for crediting contributions (we get that via git). See: https://github.com/DefinitelyTyped/DefinitelyTyped#definition-owners

File: types/ramda/test/modify-tests.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+() => {
+    const person = { name: 'James', age: 20, pets: ['dog', 'cat'] };
+    R.modify('pets', R.append('turtle'))(person); // => {name: 'James', age: 20, pets: ['dog', 'cat', 'turtle']}
+};
+
+() => {
+    const arr = [1, 2, 3];
+    R.modify(0, R.add(2), arr); // => [3, 2, 3]
+};
```
Comment (NEWESTERS) [inline types/ramda/test/modify-tests.ts:16]: I think there should be test cases for negative scenarios

Comment (adajoy) [reply types/ramda/test/modify-tests.ts:16]: the first parameter indicates property of an object, not index of an array. so if I pass -1 as first parameter, the returned array would remain the same. should I write the case?

File: types/ramda/index.d.ts (hunk 1 above)
Comment (adajoy) [reply types/ramda/index.d.ts]: removed

**Review by sheetalkamat — APPROVED** [review]

**Comment by adajoy** [comment]
Ready to merge

**Pull request merged**